src/source-pcap-folder.c

```diff
@@ -0,0 +1,943 @@
+/* Copyright (C) 2007-2016 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "suricata.h"
+#include "decode.h"
+#include "packet-queue.h"
+#include "threads.h"
+#include "threadvars.h"
+#include "tm-queuehandlers.h"
+#include "source-pcap-folder.h"
+#include "util-time.h"
+#include "util-debug.h"
+#include "conf.h"
+#include "util-error.h"
+#include "util-privs.h"
+#include "tmqh-packetpool.h"
+#include "tm-threads.h"
+#include "util-optimize.h"
+#include "flow-manager.h"
+#include "util-profiling.h"
+#include "runmode-unix-socket.h"
+#include "util-checksum.h"
+#include "util-atomic.h"
+
+#ifdef __SC_CUDA_SUPPORT__
+
+#include "util-cuda.h"
+#include "util-cuda-buffer.h"
+#include "util-mpm-ac.h"
+#include "util-cuda-handlers.h"
+#include "detect-engine.h"
+#include "detect-engine-mpm.h"
+#include "util-cuda-vars.h"
+
+#endif /* __SC_CUDA_SUPPORT__ */
+
+#include <dirent.h> // dirent
+#include <limits.h> // PATH_MAX
+#include <sys/types.h>  // stat
+#include <sys/stat.h>   // stat
+#include <unistd.h>     // stat
+
+#ifdef __linux__
+#  include <linux/version.h>
+#  if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,21)
+#    define PCAP_FOLDER_ENABLE_MONITOR
+#    include <sys/inotify.h> // inotify
+#  endif
+#endif
+#include "util-buffer.h"
+
+extern int max_pending_packets;
+
+typedef struct PcapFolderGlobalVars_ {
```

Comment:
there is a lot of duplication of code here. I think this could be sharing a lot with the exiting pcap-file support.